lib/ansible/cli/galaxy.py

```diff
@@ -1400,6 +1401,10 @@ def execute_list_collection(self, artifacts_manager=None):
         default_collections_path = AnsibleCollectionConfig.collection_paths
         collections_in_paths = {}
 
+        if context.CLIARGS.get('basedir', None):
+            collections_search_paths.update([context.CLIARGS['basedir']])
+            default_collections_path = [context.CLIARGS['basedir']] + default_collections_path
```

Comment:
The `basedir` path will need to be bytes to avoid unicode errors. Something like this:

```suggestion
            b_basedir = to_bytes(context.CLIARGS['basedir'])
            collections_search_paths.update([b_basedir])
            default_collections_path = [b_basedir] + default_collections_path
```

I did not test this. I'm just guessing from looking at the test failure.